sycl/source/detail/jit_compiler.hpp

```diff
@@ -61,8 +61,6 @@ class jit_compiler {
 
   // Manages the lifetime of the PI structs for device binaries.
   std::vector<DeviceBinariesCollection> JITDeviceBinaries;
-
-  std::unique_ptr<::jit_compiler::JITContext> MJITContext;
```

Comment:
Question: Do we need all this changes or simply forward declaring `JITContext` here and defining it in the implementation would fit our purpose? We only need to not expose `std` symbols, right?

Reply:
Forward declaration with a `unique_ptr` would only work, if it was possible to include `JITContext.h` in `jit_compiler.cpp`, which might again lead to ABI issues.

I also think that from a design point of view, it makes sense to pull the JIT context into the JIT library at this point. The explicit management of the JIT context by the SYCL runtime never materialized, so I think it is consequent to hide that detail inside the library. 

If in the future we ever need a way to manage the JIT context from the outside (e.g., clear it), we can always add functions to the interface to allow management from the outside.